libs/providers/ofrep-web/src/lib/ofrep-web-provider.ts

```diff
@@ -0,0 +1,347 @@
+   * and update the cache.
+   * @param oldContext - the old evaluation context
+   * @param newContext - the new evaluation context
+   */
+  async onContextChange(oldContext: EvaluationContext, newContext: EvaluationContext): Promise<void> {
+    try {
+      this._context = newContext;
+
+      const now = new Date();
+      if (this._retryPollingAfter !== undefined && this._retryPollingAfter > now) {
+        // we do nothing because we should not call the endpoint
+        return;
+      }
+
+      await this._evaluateFlags(newContext);
+    } catch (error) {
+      if (error instanceof OFREPApiTooManyRequestsError) {
+        this.events?.emit(ClientProviderEvents.Stale, { message: `${error.name}: ${error.message}` });
+        return;
+      }
+
+      if (
+        error instanceof OpenFeatureError ||
+        error instanceof OFREPApiFetchError ||
+        error instanceof OFREPApiUnauthorizedError ||
+        error instanceof OFREPForbiddenError
+      ) {
+        this.events?.emit(ClientProviderEvents.Error, { message: `${error.name}: ${error.message}` });
+        return;
+      }
+      this.events?.emit(ClientProviderEvents.Error, { message: `Unknown error: ${error}` });
+    }
+  }
+
+  /**
+   * onClose is called when the provider is closed, it will stop the polling if it is enabled.
+   */
+  onClose?(): Promise<void> {
+    this.stopPolling();
+    return Promise.resolve();
+  }
+
+  /**
+   * _evaluateFlags is a function that will call the bulk evaluate flags endpoint to get the flags values.
+   * @param context - the context to use for the evaluation
+   * @private
+   * @returns EvaluationStatus if the evaluation the API returned a 304, 200.
+   * @throws TargetingKeyMissingError if the API returned a 400 with the error code TargetingKeyMissing
+   * @throws InvalidContextError if the API returned a 400 with the error code InvalidContext
+   * @throws ParseError if the API returned a 400 with the error code ParseError
+   * @throws GeneralError if the API returned a 400 with an unknown error code
+   */
+  private async _evaluateFlags(context?: EvaluationContext | undefined): Promise<EvaluateFlagsResponse> {
+    try {
+      const evalReq: EvaluationRequest = {
+        context,
+      };
+      const options: RequestOptions = {
+        headers: new Headers({
+          'Content-Type': 'application/json',
+        }),
+        ...(this._etag !== null ? { headers: { 'If-None-Match': this._etag } } : {}),
+      };
+
+      const response = await this._ofrepAPI.postBulkEvaluateFlags(evalReq, options);
+      if (response.httpStatus === 304) {
+        // nothing has changed since last time, we are doing nothing
+        return { status: BulkEvaluationStatus.SUCCESS_NO_CHANGES, flags: [] };
+      }
+
+      if (response.httpStatus === 400) {
+        handleEvaluationError(response);
+      }
+
+      if (response.httpStatus === 200) {
```

Comment:
If you change 
 ```ts
      if (response.httpStatus === 400) {
        handleEvaluationError(response);
      }
```
to 
```
 ```ts
      if (response.httpStatus !== 200) {
        handleEvaluationError(response);
      }
```

you can remove the check for status 200 and do not have to throw the error in the end.
```